src/modules/Library/features/Applet/Item/Item.test.tsx

```diff
@@ -0,0 +1,119 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import { FormProvider, useForm } from 'react-hook-form';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { renderWithProviders } from 'shared/utils';
+import { mockedAppletId } from 'shared/mock';
+import { page } from 'resources';
+import * as reduxHooks from 'redux/store/hooks';
+import { library } from 'redux/modules';
+
+import { Item } from './Item';
+import { AppletUiType, LibraryForm } from '../Applet.types';
+
+const dataTestid = 'activity-item';
+const mockDispatch = jest.fn();
+
+jest.mock('./Item.styles', () => ({
+  ...jest.requireActual('./Item.styles'),
+  StyledMdPreview: ({ modelValue, 'data-testid': dataTestid }) => (
+    <div data-testid={dataTestid}>{modelValue}</div>
+  ),
+}));
+
+const FormComponent = ({ children }) => {
+  const methods = useForm<LibraryForm>({
+    defaultValues: { [mockedAppletId]: [] },
+    mode: 'onChange',
+  });
+
+  return <FormProvider {...methods}>{children}</FormProvider>;
+};
+
+describe('Item', () => {
+  beforeEach(() => {
+    jest.spyOn(reduxHooks, 'useAppDispatch').mockReturnValue(mockDispatch);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('renders checkbox and header', async () => {
+    const setAddToBuilderBtnDisabledMock = jest.spyOn(
```

Comment:
could you please move outside the setup of component (50-85 rows) into a function ?